cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -871,6 +896,11 @@ private NamespaceId validateNamespace(String namespaceId) throws BadRequestExcep
     }
     return namespace;
   }
+  
+  private String getLatestAppVersion(ApplicationId applicationId) throws Exception {
```

Comment:
Why this is throwing Exception in the method signature?

Reply:
changed the signature - it throws  ApplicationNotFoundException, IOException from `applicationLifecycleService.getLatestAppDetail`